src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1537,7 +1537,59 @@ private static Diagnostic GetFilteredDiagnostic(Diagnostic diagnostic, Compilati
             return (allAnalyzerActions, unsuppressedAnalyzers);
         }
 
-        public bool HasSymbolStartedActions => this.AnalyzerActions.SymbolStartActionsCount > 0;
+        public bool HasSymbolStartedActions(AnalysisScope analysisScope)
```

Comment:
>HasSymbolStartedActions [](start = 20, length = 23)

Can we add tests that exercise this code and demonstrate its effectiveness (ie. analyzer is run or not)?

Reply:
Sure, I can add a unit test.

Reply:
@jcouv I added unit test with https://github.com/dotnet/roslyn/pull/37303/commits/7fa3bbaad620376da108721f79623b0652e904f2 and verified it fails without my fix as the prior code also executes the non-requested analyzer.